drivers/timer/stm32_lptim_timer.c

```diff
@@ -49,66 +72,117 @@ static const struct device *const clk_ctrl = DEVICE_DT_GET(STM32_CLOCK_CONTROL_N
  * Assumptions and limitations:
  *
  * - system clock based on an LPTIM instance, clocked by LSI or LSE
- * - prescaler is set to 1 (LL_LPTIM_PRESCALER_DIV1 in the related register)
+ * - prescaler is set to a 2^value from 1 (division of the LPTIM source clock by 1)
+ *   to 128 (division of the LPTIM source clock by 128)
  * - using LPTIM AutoReload capability to trig the IRQ (timeout irq)
  * - when timeout irq occurs the counter is already reset
  * - the maximum timeout duration is reached with the lptim_time_base value
- * - with prescaler of 1, the max timeout (lptim_time_base) is 2seconds
+ * - with prescaler of 1, the max timeout (LPTIM_TIMEBASE) is 2 seconds:
+ *    0xFFFF / (LSE freq (32768Hz) / 1)
+ * - with prescaler of 128, the max timeout (LPTIM_TIMEBASE) is 256 seconds:
+ *    0xFFFF / (LSE freq (32768Hz) / 128)
+ */
+#define LPTIM_TIMEBASE CONFIG_STM32_LPTIM_TIMEBASE
+
+/* The prescaler given by the DTS and to apply to the lptim1 clock */
+#define LPTIM_CLOCK_RATIO DT_PROP(DT_NODELABEL(lptim1), st_prescaler)
+
+/* Actual lptim clock freq when the clock source is reduced by the prescaler */
+#define LPTIM_CLOCK (CONFIG_STM32_LPTIM_CLOCK / LPTIM_CLOCK_RATIO)
```

Comment:
Should use lptim_clock_freq

Reply:
removed constant, but _lptim_clock_freq/LPTIM_CLOCK_RATIO_ is used though